Scala codegen not working with local Alloy snapshots, partially breaks bloopInstall

Works:

```
cs launch com.disneystreaming.smithy4s:smithy4s-codegen-cli_2.13:0.18.33 -- generate --dependencies com.disneystreaming.alloy:alloy-protocol-tests:0.3.17 --skip openapi
```

Doesn't work:

```
# go to alloy
cd alloy
touch foo
./mill __.publishLocal

# go anywhere
cd $(mktemp -d)

cs launch com.disneystreaming.smithy4s:smithy4s-codegen-cli_2.13:0.18.33 -- generate --dependencies com.disneystreaming.alloy:alloy-protocol-tests:dev-SNAPSHOT --skip openapi
```

<details><summary>Output</summary>

```
software.amazon.smithy.model.validation.ValidatedResultException: Result contained ERROR severity validation events: 
[DANGER] -: Syntactic shape ID `awsJson1_0` does not resolve to a valid shape ID: `aws.protocols#awsJson1_0`. Did you mean to quote this string? Are you missing a model file? | SyntacticShapeIdTarget jar:file:///Users/kubukoz/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/software/amazon/smithy/smithy-aws-protocol-tests/1.56.0/smithy-aws-protocol-tests-1.56.0.jar!/META-INF/smithy/awsJson1_0/content-type.smithy:22:19
[DANGER] -: Syntactic shape ID `awsJson1_0` does not resolve to a valid shape ID: `aws.protocols#awsJson1_0`. Did you mean to quote this string? Are you missing a model file? | SyntacticShapeIdTarget jar:file:///Users/kubukoz/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/software/amazon/smithy/smithy-aws-protocol-tests/1.56.0/smithy-aws-protocol-tests-1.56.0.jar!/META-INF/smithy/awsJson1_0/defaults.smithy:153:19
(...many more of this...)

[ERROR] aws.protocoltests.ec2#AwsEc2: Unable to resolve trait `aws.api#service`. If this is a custom trait, then it must be defined before it can be used in a model. | Model.UnresolvedTrait jar:file:///Users/kubukoz/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/software/amazon/smithy/smithy-aws-protocol-tests/1.56.0/smithy-aws-protocol-tests-1.56.0.jar!/META-INF/smithy/ec2Query/main.smithy:38:1
[ERROR] aws.protocoltests.ec2#AwsEc2: Unable to resolve trait `aws.auth#sigv4`. If this is a custom trait, then it must be defined before it can be used in a model. | Model.UnresolvedTrait jar:file:///Users/kubukoz/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/software/amazon/smithy/smithy-aws-protocol-tests/1.56.0/smithy-aws-protocol-tests-1.56.0.jar!/META-INF/smithy/ec2Query/main.smithy:40:1
(...many more of this...)

[WARNING] -: Use statement refers to undefined shape: aws.api#service | Model jar:file:///Users/kubukoz/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/software/amazon/smithy/smithy-aws-protocol-tests/1.56.0/smithy-aws-protocol-tests-1.56.0.jar!/META-INF/smithy/awsJson1_0/main.smithy:5:5
[WARNING] -: Use statement refers to undefined shape: aws.api#service | Model jar:file:///Users/kubukoz/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/software/amazon/smithy/smithy-aws-protocol-tests/1.56.0/smithy-aws-protocol-tests-1.56.0.jar!/META-INF/smithy/awsJson1_1/main.smithy:5:5
[WARNING] -: Use statement refers to undefined shape: aws.api#service | Model jar:file:///Users/kubukoz/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/software/amazon/smithy/smithy-aws-protocol-tests/1.56.0/smithy-aws-protocol-tests-1.56.0.jar!/META-INF/smithy/awsJson1_1/services/machinelearning.smithy:5:5
(...many more of this...)

	at software.amazon.smithy.model.validation.ValidatedResult.validate(ValidatedResult.java:134)
	at software.amazon.smithy.model.validation.ValidatedResult.unwrap(ValidatedResult.java:121)
	at smithy4s.codegen.internals.ModelLoader$.load(ModelLoader.scala:83)
	at smithy4s.codegen.internals.CodegenImpl$.generate(CodegenImpl.scala:46)
	at smithy4s.codegen.Codegen$.generate(Codegen.scala:22)
	at smithy4s.codegen.Codegen$.generateToDisk(Codegen.scala:25)
	at smithy4s.codegen.cli.Main$.$anonfun$main$1(Main.scala:46)
	at smithy4s.codegen.cli.Main$.$anonfun$main$1$adapted(Main.scala:44)
	at scala.util.Either.map(Either.scala:390)
	at smithy4s.codegen.cli.Main$.main(Main.scala:44)
	at smithy4s.codegen.cli.Main.main(Main.scala)

```

</details>

The openapi skip is there because that particular module doesn't work in there anyway. If you don't add `--skip openapi`, this is what you get:

```
software.amazon.smithy.openapi.OpenApiException: The `httpPrefixHeaders` trait is not supported by OpenAPI and was found on `aws.protocoltests.restjson#HttpEmptyPrefixHeadersInput$prefixHeaders`
	at software.amazon.smithy.openapi.fromsmithy.mappers.CheckForPrefixHeaders.check(CheckForPrefixHeaders.java:67)
	at software.amazon.smithy.openapi.fromsmithy.mappers.CheckForPrefixHeaders.lambda$before$1(CheckForPrefixHeaders.java:44)
	at java.base/java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1715)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
	at software.amazon.smithy.openapi.fromsmithy.mappers.CheckForPrefixHeaders.before(CheckForPrefixHeaders.java:43)
	at software.amazon.smithy.openapi.fromsmithy.OpenApiMapper$1.before(OpenApiMapper.java:407)
	at software.amazon.smithy.openapi.fromsmithy.OpenApiConverter.convertWithEnvironment(OpenApiConverter.java:324)
	at software.amazon.smithy.openapi.fromsmithy.OpenApiConverter.convertToNode(OpenApiConverter.java:155)
	at alloy.openapi.package$.$anonfun$convertWithConfig$6(package.scala:85)
	at scala.collection.StrictOptimizedIterableOps.map(StrictOptimizedIterableOps.scala:100)
	at scala.collection.StrictOptimizedIterableOps.map$(StrictOptimizedIterableOps.scala:87)
	at scala.collection.immutable.Set$Set1.map(Set.scala:165)
	at alloy.openapi.package$.$anonfun$convertWithConfig$4(package.scala:78)
	at scala.collection.immutable.List.flatMap(List.scala:294)
	at alloy.openapi.package$.convertWithConfig(package.scala:75)
	at smithy4s.codegen.internals.CodegenImpl$.generate(CodegenImpl.scala:96)
	at smithy4s.codegen.Codegen$.generate(Codegen.scala:22)
	at smithy4s.codegen.Codegen$.generateToDisk(Codegen.scala:25)
	at smithy4s.codegen.cli.Main$.$anonfun$main$1(Main.scala:46)
	at smithy4s.codegen.cli.Main$.$anonfun$main$1$adapted(Main.scala:44)
	at scala.util.Either.map(Either.scala:390)
	at smithy4s.codegen.cli.Main$.main(Main.scala:44)
	at smithy4s.codegen.cli.Main.main(Main.scala)
```

Impact: for smithy4s devs, this can break things like `bloopInstall`, because the `alloy-protocol-tests` module is used in a resource generator for http4s tests. Currently, it doesn't _actually_ break because somehow the `bloopInstall` task ignores the error.

--- a/build.sbt
+++ b/build.sbt
@@ -448,7 +448,7 @@ lazy val codegen = projectMatrix
       Dependencies.Circe.generic.value,
       Dependencies.collectionsCompat.value,
       "org.scala-lang" % "scala-reflect" % scalaVersion.value,
-      "io.get-coursier" %% "coursier" % "2.1.9"
+      "io.get-coursier" %% "coursier" % "2.1.24"
     ),
     libraryDependencies ++= munitDeps.value,
     scalacOptions := scalacOptions.value

